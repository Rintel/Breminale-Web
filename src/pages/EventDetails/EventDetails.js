import React from 'react';
import PropTypes from 'prop-types';
import Header from '../../components/Header/Header';
import { CardMedia, CardTitle } from 'material-ui/Card';
import axios from 'axios';
import moment from 'moment';

moment.locale('de');

const propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.node,
    }).isRequired,
  }).isRequired
};

export default class EventDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      content: {
        name: null,
        description: null,
        start_time: '2016-07-15T18:00:00.000+02:00',
      }
    };
  }

  componentDidMount() {
    axios.get(`https://serene-ocean-3356.herokuapp.com/api/v1/events/${this.props.match.params.id}.json`)
      .then(res => {
        const content = res.data;
        this.setState({ content: content });
    });
  }

  render() {
  return (
      <div className='content-container'>
        <Header title={'Breminale-Web'} />
        <div className='event-container'>
          <div className='event-banner'>
            <CardMedia
              overlay={<CardTitle title={this.state.content.name} subtitle={moment(this.state.content.start_time).format('LLL')} />}
            >
              <img src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg" alt="" />
            </CardMedia>
          </div>
          <div className='event-description'>
            <div className='event-description__header'>
              {'Beschreibung'}
            </div>
            <div className='event-description__content'>
              <p>{this.state.content.description}</p>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

EventDetails.propTypes = propTypes;
