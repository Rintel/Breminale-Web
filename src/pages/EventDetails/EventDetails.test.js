import React from 'react';
import { shallow } from 'enzyme';
import EventDetails from './EventDetails';


const baseProps = {
  match: {
    params: {
      id: 321
    }
  }
};

describe('EventDetails Component is rendering', () => {
  test('valid props are passed', () => {
    const wrapper = shallow(
      <EventDetails {...baseProps} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
