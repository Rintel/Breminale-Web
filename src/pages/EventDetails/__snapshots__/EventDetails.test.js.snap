// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails Component is rendering valid props are passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventDetails
    match={
      Object {
        "params": Object {
          "id": 321,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          title="Breminale-Web"
        />,
        <div
          className="event-container"
        >
          <div
            className="event-banner"
          >
            <CardMedia
              overlay={
                <CardTitle
                  subtitle="15. Juli 2016 18:00"
                  title={null}
                />
              }
            >
              <img
                alt=""
                src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
              />
            </CardMedia>
          </div>
          <div
            className="event-description"
          >
            <div
              className="event-description__header"
            >
              Beschreibung
            </div>
            <div
              className="event-description__content"
            >
              <p />
            </div>
          </div>
        </div>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "title": "Breminale-Web",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="event-banner"
            >
              <CardMedia
                overlay={
                  <CardTitle
                    subtitle="15. Juli 2016 18:00"
                    title={null}
                  />
                }
              >
                <img
                  alt=""
                  src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                />
              </CardMedia>
            </div>,
            <div
              className="event-description"
            >
              <div
                className="event-description__header"
              >
                Beschreibung
              </div>
              <div
                className="event-description__content"
              >
                <p />
              </div>
            </div>,
          ],
          "className": "event-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <CardMedia
                overlay={
                  <CardTitle
                    subtitle="15. Juli 2016 18:00"
                    title={null}
                  />
                }
              >
                <img
                  alt=""
                  src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                />
              </CardMedia>,
              "className": "event-banner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt=""
                  src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                />,
                "overlay": <CardTitle
                  subtitle="15. Juli 2016 18:00"
                  title={null}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "http://www.joko-bremen.de/breminale15/SAM_1657.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="event-description__header"
                >
                  Beschreibung
                </div>,
                <div
                  className="event-description__content"
                >
                  <p />
                </div>,
              ],
              "className": "event-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Beschreibung",
                  "className": "event-description__header",
                },
                "ref": null,
                "rendered": "Beschreibung",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <p />,
                  "className": "event-description__content",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            title="Breminale-Web"
          />,
          <div
            className="event-container"
          >
            <div
              className="event-banner"
            >
              <CardMedia
                overlay={
                  <CardTitle
                    subtitle="15. Juli 2016 18:00"
                    title={null}
                  />
                }
              >
                <img
                  alt=""
                  src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                />
              </CardMedia>
            </div>
            <div
              className="event-description"
            >
              <div
                className="event-description__header"
              >
                Beschreibung
              </div>
              <div
                className="event-description__content"
              >
                <p />
              </div>
            </div>
          </div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "title": "Breminale-Web",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="event-banner"
              >
                <CardMedia
                  overlay={
                    <CardTitle
                      subtitle="15. Juli 2016 18:00"
                      title={null}
                    />
                  }
                >
                  <img
                    alt=""
                    src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                  />
                </CardMedia>
              </div>,
              <div
                className="event-description"
              >
                <div
                  className="event-description__header"
                >
                  Beschreibung
                </div>
                <div
                  className="event-description__content"
                >
                  <p />
                </div>
              </div>,
            ],
            "className": "event-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <CardMedia
                  overlay={
                    <CardTitle
                      subtitle="15. Juli 2016 18:00"
                      title={null}
                    />
                  }
                >
                  <img
                    alt=""
                    src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                  />
                </CardMedia>,
                "className": "event-banner",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <img
                    alt=""
                    src="http://www.joko-bremen.de/breminale15/SAM_1657.jpg"
                  />,
                  "overlay": <CardTitle
                    subtitle="15. Juli 2016 18:00"
                    title={null}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "http://www.joko-bremen.de/breminale15/SAM_1657.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="event-description__header"
                  >
                    Beschreibung
                  </div>,
                  <div
                    className="event-description__content"
                  >
                    <p />
                  </div>,
                ],
                "className": "event-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Beschreibung",
                    "className": "event-description__header",
                  },
                  "ref": null,
                  "rendered": "Beschreibung",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p />,
                    "className": "event-description__content",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
